phoenix-core/src/main/java/org/apache/phoenix/query/GuidePostsCache.java

```diff
@@ -85,39 +106,54 @@ public GuidePostsCache(ConnectionQueryServices queryServices, Configuration conf
                 })
                 // Log removals at TRACE for debugging
                 .removalListener(new PhoenixStatsCacheRemovalListener())
-                // Automatically load the cache when entries are missing
-                .build(isStatsEnabled ? new StatsLoader() : new EmptyStatsLoader());
+                // Automatically load the cache when entries need to be refreshed
+                .build(cacheLoader);
     }
 
     /**
-     * {@link CacheLoader} implementation for the Phoenix Table Stats cache.
+     * {@link PhoenixStatsLoader} implementation for the Stats Loader.
      */
-    protected class StatsLoader extends CacheLoader<GuidePostsKey, GuidePostsInfo> {
+    protected class StatsLoaderImpl implements PhoenixStatsLoader {
         @Override
-        public GuidePostsInfo load(GuidePostsKey statsKey) throws Exception {
+        public boolean needsLoad() {
+            // Whenever it's called, we try to load stats from stats table
+            // no matter it has been updated or not.
+            // TODO:
```

Comment:
Can you link a JIRA for this todo?

Reply:
I'll move this TODO because here I just want to point out the possible optimizations we can do here which haven't been formalized as the solutions we must do in the future.